### ISSUE TITLE
Fumble_Yardage variable

### ISSUE THREAD

**Issue by TheMathNinja** [mrcaseb/nflfastR#41]
I was wondering if we could get a fumble_yardage variable included in this dataset? I'm looking at a play like game_id 2019_01_BUF_NYJ, play_id 965 where Sam Darnold botches a snap and fumbles out of bounds for -3 yards. He gets credited with -3 fumble yards on this play by the NFL but I don't see this stat anywhere in the nflfastR dataset. Could we get this one included?

**Comment by mrcaseb** [comment]
This is `statId == 57`. And we are currently not using the yards for this ID. Good catch. I think we could and should add that.

**Comment by TheMathNinja** [comment]
Because I'm a little less familiar with the deeper guts of this stuff, how might I go about looking up the various statId's myself? I'd love to get acquainted with the list of these and their descriptions and uses.

**Comment by mrcaseb** [comment]
You can check [this list](http://www.nflgsis.com/gsis/Documentation/Partners/StatIDs.html) to learn which IDs are being used. In the raw underlying data those are used in the `playStats` data frame for each play.

**Comment by mrcaseb** [comment]
nflfastR uses [this function](https://github.com/mrcaseb/nflfastR/blob/master/R/helper_tidy_play_stats.R) to summarize all stats in a single row. And there we don't use the yardage for stat ID 57. Most of the logic in there came from nflscrapR although we already fixed some bugs. Using the yardage for 57 was never implemented and nobody cared for years but it seems justified

**Comment by guga31bb** [comment]
Was curious about what actually happened on this play (how do you botch a snap out of bounds) and the play description seems...misleading https://twitter.com/benbbaldwin/status/1283207642224177155

But yes we can add the stat 57

**Comment by mrcaseb** [comment]
I have looked for stat ID 57 (as well as 58, 61, 62) in the 2011+ data. It is reproducible with the following code

```r
library(tidyverse)
library(nflfastR)

games <- fast_scraper_schedules(2011:2019)

check_cols <- function(id) {
  season <- substr(id, 1, 4)
  path <- "https://raw.githubusercontent.com/guga31bb/nflfastR-raw/master/raw"
  raw_data <- readRDS(glue::glue("{path}/{season}/{id}.rds"))
  plays <- raw_data$data$viewer$gameDetail$plays
  stats <- tidyr::unnest(plays %>% dplyr::select(-"yards"), cols = c("playStats")) %>%
    dplyr::mutate(
      game_id = as.character(id),
      yards = as.integer(.data$yards),
      statId = as.numeric(.data$statId),
      team.abbreviation = as.character(.data$team.abbreviation)
    ) %>%
    dplyr::rename(
      player.esbId = "gsisPlayer.id",
      player.displayName = "playerName",
      teamAbbr = "team.abbreviation"
    ) %>%
    dplyr::select(
      "game_id",
      "playId",
      "statId",
      "yards",
      "teamAbbr",
      "player.displayName"
    ) %>% 
    filter(statId %in% c(57, 58, 61, 62))
  return(stats)
}

future::plan("multiprocess")
all <- furrr::future_map_dfr(games$game_id, check_cols)

all %>% as_tibble()

# A tibble: 733 x 6
   game_id         playId statId yards teamAbbr player.displayName
   <chr>            <int>  <dbl> <int> <chr>    <chr>             
 1 2011_01_DET_TB     644     57    -4 DET      M.Stafford        
 2 2011_01_DET_TB    4277     57    -4 TB       K.Winslow         
 3 2011_01_DET_TB    4277     57    -8 TB       P.Parker          
 4 2011_01_CAR_ARI   1509     57    -8 ARI      K.Kolb            
 5 2011_01_SEA_SF    3065     57    -1 SEA      T.Jackson         
 6 2011_01_NE_MIA    2896     57    -4 MIA      C.Henne           
 7 2011_02_STL_NYG   1369     57   -10 STL      S.Bradford        
 8 2011_03_JAX_CAR   2561     57   -11 JAC      M.Turk            
 9 2011_03_JAX_CAR   2769     57    -2 JAC      B.Gabbert         
10 2011_03_JAX_CAR   3006     57    -4 JAC      B.Gabbert         
# ... with 723 more rows

all %>% group_by(statId) %>% summarise(n = n(), sum_yards = sum(yards))

# A tibble: 3 x 3
  statId     n sum_yards
   <dbl> <int>     <int>
1     57   724     -4536
2     61     8        71
3     62     1        70
```

There are a lot of QB plays, like the above mentioned, having stat id 57 (most likely they are using it for aborted snaps). However I kind of dislike the variable name `fumble_yards` because it is somewhat misleading. Any suggestions how we can name the variable?

**Comment by TheMathNinja** [comment]
Sadly I cannot duplicate that code because I get the error message: "Error in library(nflfastR) : there is no package called ‘nflfastR’" I've been doing all my analysis from the data repo.

I see now that this is sort of a double-fix with issue #27. The tricky part of this is that in the cases I brought up in that issue, it made sense to call the variable "lateral_yards" because it was yards gained having received a lateral. In the majority of cases, it seems like the player gaining/losing yards here is the QB and the "lateral" is actually the snap. On the Darnold play, he gets -3 because he fumbled the ball out of bounds (which itself was actually a lateral). 

All this to say: I know that the stats provider for MyFantasyLeague actually includes stat ID 57 under the category heading of "Own Fumble Recovery Yardage" along with stat IDs 55, 56, and 58. They are all listed in that category. That's one approach you could take.

NFL Guide for Statisticians actually calls it "fumble yardage" in the case of aborted plays, which might be why MFL includes it with fumble recoveries. I know nflfastR already uses "lateral_recovery" as a variable, so "lateral_recovery_yards" makes sense, especially if we can add "lateral_recovery_player". But like Ben brought up in Issue 27, I'm not sure how many columns are needed for how many unique players might receive stat 57 on a given play.

I could say much more intelligent things if I could reproduce these plays like you have.

**Comment by mrcaseb** [comment]
If you want to reproduce without installing nflfastR you can replace the games line 4 with 

```r
games <- readRDS(url("https://github.com/leesharpe/nfldata/blob/master/data/games.rds?raw=true")) %>%
  filter(between(season, 2011, 2019))
```

I am on the smartphone now and can't check it but this should work.
